pkg/oc/cli/cmd/create/route.go

```diff
@@ -156,6 +156,10 @@ func CreateEdgeRoute(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, ar
 	dryRun := kcmdutil.GetFlagBool(cmd, "dry-run")
 	actualRoute := route
 
+	if kcmdutil.GetFlagString(cmd, "output") != "" {
+		kcmdutil.PrintObject(cmd, actualRoute, out)
```

Comment:
We can choose how it runs with oc but the helptext does not suggest that --output could only be used in the dry-run case. Using 'oc create -f hello-pod.json -o yaml' prints the yaml of the object, I personally don't think that is useful but shouldn't we try to make the create commands act similarly?

Reply:
As far as I can tell, the dryRun case below this currently does not short-circuit, which means that (regardless of --dry-run being specified or not), we would end up printing the yaml/json object as well as the "successful" output right below it.

By placing this entire block below the "dryrun" one, we could add a short-circuit if the "output" condition is met, to prevent both printing cases from happening